lib/trino-filesystem/src/main/java/io/trino/filesystem/hdfs/HdfsFileSystem.java

```diff
@@ -75,6 +83,30 @@ public void deleteFile(String path)
         });
     }
 
+    @Override
+    public void deleteFiles(Collection<String> paths)
+            throws IOException
+    {
+        Map<Path, List<Path>> pathsGroupedByDirectory = paths.stream().collect(
+                groupingBy(path -> hadoopPath(path.replaceFirst("/[^/]*$", "")),
+                        mapping(HadoopPaths::hadoopPath, toList())));
```

Comment:
style: put each arg of `groupingBy` on its own line
```suggestion
                groupingBy(
                        path -> hadoopPath(path.replaceFirst("/[^/]*$", "")),
                        mapping(HadoopPaths::hadoopPath, toList())));
```